File: Solutions/0066/0066.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2021/1/12 09:39
# File: 0066.py
# Desc: 
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        more = 0
        for i in range(len(digits)-1, -1, -1):
            num = digits[i]
            if num == 9: 
                more = 1
                digits[i] = 0
            else: 
                digits[i] = num + 1
                return digits
        if more:
            digits.insert(0, 1)
        return digits

File: Solutions/0066/0066.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n = digits.end() - digits.begin();
        int p = 1;
        for (int i=n-1; i>=0; i--){
            if (digits[i] == 9){
                digits[i] = 0;
                p = 1;
            }else {
                digits[i]++;
                p = 0;
            }
            if (p == 0) break;
        }
        if (p) digits.insert(digits.begin(), 1);
        return digits;
    }
};
